debian/rules

```diff
@@ -2,7 +2,6 @@
 
```

Comment:
We should bump epoch due to #134. If it is solved somehow, I don't see how. Please, clarify.

Reply:
Epoch for debian was already bumped [here](https://github.com/tarantool/http/commit/4db9aa3a16a767fd82b2125984af16c262b3ec6b#diff-c0279f62819ca56bac5a167034ae765318b020d36c8b151f8d999ba064c5ee39R4)

Reply:
AFAIR, packpack replaces `Version` field in `debian/control` entirely. I guess we should teach packpack to replace just what is after the colon.

Am I wrong here?

Reply:
I did some checks and what I get:

first, I build `tarantool-http` package using `packpack`

```bash
OS=ubuntu DIST=focal  ./packpack/packpack
```
then I run docker container to check which version of `tarantool-http` is installed

```bash
docker run --rm -it -v $(pwd)/build:/build ubuntu:focal bash
...
root@a3b036db23df:/# apt update && apt install -y curl
root@a3b036db23df:/# curl -LsSf https://www.tarantool.io/release/1.10/installer.sh |  bash
root@a3b036db23df:/# apt install -y tarantool tarantool-dev
root@a3b036db23df:/# cd build/
root@a3b036db23df:/build# dpkg -i tarantool-http_1.2.0.11-1_amd64.deb 
root@a3b036db23df:/build# apt-cache policy tarantool-http
tarantool-http:
  Installed: 1.2.0.11-1
  Candidate: 1:1.1.0.0-1
  Version table:
     1:1.1.0.0-1 500
        500 https://download.tarantool.org/tarantool/release/1.10/ubuntu focal/main amd64 Packages
        500 https://download.tarantool.org/tarantool/modules/ubuntu focal/main amd64 Packages
 *** 1.2.0.11-1 100
        100 /var/lib/dpkg/status
```

Okay, trying to build and install the same, but with shallow git clone. Whay I get after the same operations as I did above:

```bash
root@4ca7d7f7c3d5:/build# apt-cache policy tarantool-http
tarantool-http:
  Installed: 0.0.1-1
...
```

The version of the package in it's spec is not affecting on a build using `packpack` as it gets version from `git describe`. I see a problem in the current uploaded version in the Tarantool, which contains epoch version.

Somehow `tarantool-http` version 1:1.1.0 was uploaded into repository (prod) and it has bigger priority, so I copy it into dev repo as well.

Let's try to upload tarantool-http to the test repo and see which module will be installed.

Current repo list:
```bash
cat /etc/apt/sources.list.d/tarantool_2.list
deb https://download.tarantool.org/tarantool/release/series-2/ubuntu/ focal main
deb https://hb.bizmrg.com/tarantool-dev/restored/release/modules/ubuntu/ focal main
```

Check package candidate:
```bash
root@7f0778511fce:/# apt-cache policy tarantool-http
tarantool-http:
  Installed: (none)
  Candidate: 1:1.1.0.0-1
  Version table:
     1:1.1.0.0-1 500
        500 https://hb.bizmrg.com/tarantool-dev/restored/release/modules/ubuntu focal/main amd64 Packages
     1.2.0.10-1 500
        500 https://hb.bizmrg.com/tarantool-dev/restored/release/modules/ubuntu focal/main amd64 Packages
```

Alexander, please tell me the solution to this problem, I'm stuck.

Reply:
packpack replaces the version from a repository to `git describe` based one (or to one provided in the `VERSION` environment variable) as for RPM as well as for Deb packages. AFAIS, the relevant code for Debian/Ubuntu is [here](https://github.com/packpack/packpack/blob/194f898750cbbee6dd35255fe8aa7ec878eb3836/pack/deb.mk#L106-L109).

There are different possible ways to teach packpack to accept our epoch:

1. Take care to EPOCH environment variable (we should also support it for RPMs, if we'll do it for Debs).
2. Extract `Version: (.*)` from `debian/control` and, if the verison has epoch, take care to this epoch value.
3. Both 1 and 2.

I like the second more than the first, because it allows to write the epoch into the repository once (like we do in 4db9aa3a16a767fd82b2125984af16c262b3ec6b) and call `OS=<...> DIST=<...> /path/to/packpack` as usual. It is most convenient in our case. However the first one also may be useful.

BTW, there is related issue in packpack: https://github.com/packpack/packpack/issues/79.

Reply:
It seems, it'll be fixed on the packpack side. Closing the discussion so.